Submit branch

Alinchik1 opened the pull request:
Diploma project

Alinchik1 committed:
create semantic markup for homepage

Alinchik1 committed:
export svg for homepage, create sprite, add favicon

Alinchik1 committed:
export images, convert to webp, compress and add to the html homepage

Alinchik1 committed:
link fonts, normalize, main and homepage css to homepage

Alinchik1 committed:
Merge pull request #1 from Alinchik1/create-index-html

Create index html

Alinchik1 committed:
add main styles for header and footer

Alinchik1 committed:
add styles to homepage

Alinchik1 committed:
add svg

Alinchik1 committed:
add hover state to social media links

Alinchik1 committed:
Merge pull request #2 from Alinchik1/create-styles-homepage

Create styles homepage

Alinchik1 committed:
create html code for catalog page

Alinchik1 committed:
upload images, convert to webp and compress, add svgs to sprite

Alinchik1 committed:
fix heading tree

Alinchik1 committed:
Merge pull request #3 from Alinchik1/create-catalog-html

Create catalog html

Alinchik1 committed:
create styles for catalog page

Alinchik1 committed:
add dot to the text description

Alinchik1 committed:
Merge pull request #4 from Alinchik1/add-styles-to-calatog-page

Add styles to catalog page

Alinchik1 committed:
create html for the blogpage

Alinchik1 committed:
upload, convert and compress images for blogpage html, upload svg

Alinchik1 committed:
link all html files, rename title, change sections without headings to div

Alinchik1 committed:
Merge pull request #5 from Alinchik1/create-html-blog-page

Create html blog page

Alinchik1 committed:
apply styles to blogpage, remove header padding, fix the dot in homepage

Alinchik1 committed:
add hover state, active state and separator for categories

Alinchik1 committed:
Merge pull request #6 from Alinchik1/add-styles-blogpage

Add styles blogpage

Alinchik1 committed:
create html code for contact page

Alinchik1 committed:
add sr text to buttons, add hover and focus states, fix faq, add svg to banner grid

Alinchik1 committed:
add focus state

Alinchik1 committed:
add links for contacts and container wrapers

Alinchik1 committed:
add active state, link styles, adjust px per design

Alinchik1 committed:
Merge pull request #7 from Alinchik1/create-html-contacts

Create html contacts

Alinchik1 committed:
add wraper, fix trending section and styles for homepage, add active state

Alinchik1 committed:
add wraper, adjust px, margin for blogpage

Alinchik1 committed:
add padding to footer

Alinchik1 committed:
add wraper, range input, adjust styles px for catalog page

Alinchik1 committed:
add transition, adjust px for styles

Alinchik1 committed:
remove unessesary code from catalog, rename contacts title, add focus state for catalog switcher

Alinchik1 committed:
add flex-wrap property

Alinchik1 committed:
Merge pull request #8 from Alinchik1/final-styles-for-pixmart

Final styles for pixmart

Alinchik1 committed:
link js to catalog html code, fix focus state for switcher

Alinchik1 committed:
Merge pull request #9 from Alinchik1/add-java-script

link js to catalog html code, fix focus state for switcher

zzoysanna commented:
You have [3] more attempt to finish your diploma project

**GIT** 

- [x] 1. ❤️  **The site should have a GitHub repository**
- [x] 2. ❤️ **There's no system file like ```.DS_Store``` or ```Thumbs.db``` in the repository**
- [x] 3. ❤️  **There are no commits in the target branch (usually ```main```)**
- [x] 4. 💛 **Commit messages are descriptive**
- [x] 5. 💛 **Each commit is a logical unit**
- [x] 6. 💚 **Commits are atomic**
- [ ] 7. 💚 **Commit message no longer than 60 symbols**

**HTML**

- [x] 1. ❤️ **The document must begin with ```<!DOCTYPE HTML>```.**  
- [x] 2. ❤️ **Each page must have the lang attribute set according to the page's language.**  
- [x] 3. ❤️ **The document must have a specified encoding.**  
- [x] 4. ❤️ **Title Tag must be set on all pages.**  
- [ ] 5. 💚 **Description meta tag must be set** <br>
No description tags <br><br>
- [x] 6. 💚 **Favicon icons display correctly.**  
- [x] 7. ❤️ **All elements must be marked up and styled.**  
- [x] 8. ❤️ **No gross markup errors.**  
- [ ] 9. ❤️ **Semantic Markup**  <br>
The Google Maps ```<iframe>``` has no title attribute, which is critical for accessibility<br><br>
- [x] 10. ❤️ **The layout must be valid.**  
- [x] 11. 💛 **Classes are named correctly:**  
- [x] 12. 💛 **Use the minimum possible number of HTML elements.**  
- [x] 13. 💛 **The logo on internal pages should lead to the home page.**  
- [x] 14. ❤️ **All pages should be linked and checked for broken links.**  
- [ ] 15. 💛 **External links should have the attribute ```target="_blank"```.**  <br>
External links (e.g., social media links) lack target="_blank"<br><br>
- [x] 16. ❤️ **Each page must have an ```<h1>``` tag that is different from the page title.**  
- [ ] 17. 💛**The heading tree structure is maintained.**<br>
Header on catalog page is divided by two, but logically it’s one ```<h1>```<br><br>
- [x] 18. ❤️**Appropriate types for input are specified.** 
- [ ] 19. ❤️ **Each form element should have a ```<label>```.**<br>
The search bar input lacks an accessible ```<label>```<br><br>
- [x] 20. 💛 **Resizing ```<textarea>``` should not break the layout.** 
- [x] 21. ❤️ **Basic HTML form validation should be implemented.**

**CSS**

- [ ] 1. ❤️ **Single CSS File**<br>
Multiple CSS files (main.css, homepage.css, blogpage.css, catalog.css, contacts.css) are used<br><br>
- [ ] 2. 💚 **Include Normalize.css** <br>
normalize.css linked after custom slyles<br><br>
- [ ] 3. ❤️ **All fonts used in the design are connected to the pages.** <br>
On blogpage.html fonts are not properly connected - Times New Roman renders instead<br><br>
- [x] 4. ❤️ **Provide fallback fonts and family types at the end of the font list.** 
- [x] 5. ❤️ **Do not use !important in CSS.** 
- [x] 6. ❤️**Do not use #id for styling.** 
- [x] 7. 💛**Avoid nesting selectors more than two levels deep.**  
- [x] 8. 💛**Avoid styling tags directly** 
- [x] 9. 💛**Elements with background images should have a matching background color, if it's matter for text contrast** 
- [x] 10. ❤️**Use consistent units for element sizes and positioning.** 
- [ ] 11. 💛**Colors should be in a consistent format (hex or rgba).** <br />
Both hex and rgba are used<br><br>
- [x] 12. ❤️**Use flex or grid for layouts** <br />
- [x] 13. ❤️**Use appropriate properties for animations.** <br />
- [x] 14. ❤️**All element states from the style guide are implemented.** 
- [x] 15. ❤️**Links must respond to ```:hover, :active и :focus```.** <br />
- [x] 16. 💛**Interactions( ```:hover, :active и :focus```) should not change the document flow.** <br />
- [x] 17. 💛**Use vendor prefixes generated according to supported browsers.** <br />
- [ ] 18. 💛**Minify CSS files.** <br />
- [x] 19. 💛**Remove unused CSS styles.** <br />
- [x] 20. 💛**CSS should be error-free and valid.** <br />
- [ ] 21. 💚**Use CSS variables for repeated values – colors, sizes, etc.** <br>
Repeated values like colors (#0265DC, #FFFFFF) are hardcoded multiple times across files<br><br>
- [ ] 22. 💚**Comment magic numbers in functions and browser hacks.** 

**Images**

- [x] 1. ❤️ **Correct Image Format** <br />
- [x] 2. ❤️ **Extract images without style filters from the design.** <br />
- [ ] 3. ❤️ **Image Insertion** <br />
Images in product__list on index.html are decorative, but inserted with ing tag<br><br>
- [x] 4. ❤️ **Specify dimensions for all images.** <br />
- [x] 5. ❤️ **Provide alt text for all images.** <br />
- [x] 6. 💛 **Optimize images for browser rendering.** <br />
- [x] 7. 💚 **Use vector sprites where applicable.** <br />

**Other** 

- [x] 1. ❤️**The root document should have folders like css, img, js.** 
- [x] 2. ❤️ **The main page should be named index.html**
- [x] 3. ❤️ **File names and extensions should be lowercase without spaces.** <br />
- [x] 4. ❤️ **Files should be uniformly formatted.** <br />
- [ ] 5. ❤️ **The layout should match the design exactly with Perfect pixel** <br />
Layouts doesn’t match the design<br><br>
- [x] 6. ❤️ **The layout should display identically in the latest versions of Chrome, Opera, Firefox, Safari, and Edge.** 
- [ ] 7. ❤️ **The site should look good on all screen resolutions.** <br />
If width less then 1440px layout changes - scale or scroll should be used instead<br><br>
- [ ] 8. ❤️ **The site should handle content overflow gracefully.** <br />
Layout breaks in following cases: adding new items to trending-product__list, changing title in catalog-box<br><br>
- [x] 9. 💛**Unnecessary code should be removed before production.** <br />
- [ ] 10. 💚 **Use a typographer for text formatting.**  
- [x] 11. 💛 **There should be no errors in the browser console.**

Alinchik1 committed:
add title attribute to iframe

Alinchik1 committed:
add laber for search input

Alinchik1 committed:
change position for normalize and fix font for catalog page

Alinchik1 committed:
fix overflow with scroll and remove flex wrap to keep the design and layout

Alinchik1 committed:
add target="_blank" to external links

Alinchik1 committed:
change product__list img to a backgroung

Alinchik1 committed:
replace all css files with single styles.css file

Alinchik1 commented:
Good morning Zoia,

My project is ready for the second review. Fixed points from the comments:

1. The Google Maps <iframe> has no title attribute, which is critical for accessibility - title attribute added
2. External links (e.g., social media links) lack target="_blank" - target attribute added
3. The search bar input lacks an accessible <label> - label added to all HTML search bars
4. Multiple CSS files (main.css, homepage.css, blogpage.css, catalog.css, contacts.css) are used - single styles.css created
5. normalize.css linked after custom styles - normalize linked before styles.css
6. On blogpage.html fonts are not properly connected - Times New Roman renders instead - fixed in CSS - incorrect font name was used
7. Images in product__list on index.html are decorative, but inserted with img tag - img is now used as a background 
8. Layouts doesn’t match the design - fixed the with overflow scroll, removed flex-wrap to keep the layout
9. If width less then 1440px layout changes - scale or scroll should be used instead - fixed overflow with scroll
10. Layout breaks in following cases: adding new items to trending-product__list, changing title in catalog-box - fixed

Thank you.

zzoysanna commented:
You have [2] more attempt to finish your diploma project

GIT 

- [x] 1. ❤️  **The site should have a GitHub repository**
- [x] 2. ❤️ **There's no system file like ```.DS_Store``` or ```Thumbs.db``` in the repository**
- [x] 3. ❤️  **There are no commits in the target branch (usually ```main```)**
- [x] 4. 💛 **Commit messages are descriptive**
- [x] 5. 💛 **Each commit is a logical unit**
- [x] 6. 💚 **Commits are atomic**
- [ ] 7. 💚 **Commit message no longer than 60 symbols**

1. Some of messages are longer

HTML

- [x] 1. ❤️ **The document must begin with ```<!DOCTYPE HTML>```.**  
- [x] 2. ❤️ **Each page must have the lang attribute set according to the page's language.**  
- [x] 3. ❤️ **The document must have a specified encoding.**  
- [x] 4. ❤️ **Title Tag must be set on all pages.**  
- [ ] 5. 💚 **Description meta tag must be set** 

1. No description tags 

- [x] 6. 💚 **Favicon icons display correctly.**  
- [x] 7. ❤️ **All elements must be marked up and styled.**  
- [x] 8. ❤️ **No gross markup errors.**  
- [x] 9. ❤️ **Semantic Markup**  
- [x] 10. ❤️ **The layout must be valid.**  
- [x] 11. 💛 **Classes are named correctly:**  
- [ ] 12. 💛 **Use the minimum possible number of HTML elements.**  

1. Redundant containers for images in product__list (index.html)

- [x] 13. 💛 **The logo on internal pages should lead to the home page.**  
- [x] 14. ❤️ **All pages should be linked and checked for broken links.**  
- [x] 15. 💛 **External links should have the attribute ```target="_blank"```.**  
- [x] 16. ❤️ **Each page must have an ```<h1>``` tag that is different from the page title.**  
- [ ] 17. 💛**The heading tree structure is maintained.**

1. Header on catalog page is divided by two, but logically it’s one ```<h1>```

- [x] 18. ❤️**Appropriate types for input are specified.** 
- [x] 19. ❤️ **Each form element should have a ```<label>```.**
- [x] 20. 💛 **Resizing ```<textarea>``` should not break the layout.** 
- [x] 21. ❤️ **Basic HTML form validation should be implemented.**

CSS

- [x] 1. ❤️ **Single CSS File**
- [x] 2. 💚 **Include Normalize.css** 
- [x] 3. ❤️ **All fonts used in the design are connected to the pages.** 
- [x] 4. ❤️ **Provide fallback fonts and family types at the end of the font list.** 
- [x] 5. ❤️ **Do not use !important in CSS.** 
- [x] 6. ❤️**Do not use #id for styling.** 
- [x] 7. 💛**Avoid nesting selectors more than two levels deep.**  
- [x] 8. 💛**Avoid styling tags directly** 
- [x] 9. 💛**Elements with background images should have a matching background color, if it's matter for text contrast** 
- [x] 10. ❤️**Use consistent units for element sizes and positioning.** 
- [ ] 11. 💛**Colors should be in a consistent format (hex or rgba).** <br />

1. Both hex and rgba are used

- [x] 12. ❤️**Use flex or grid for layouts** <br />
- [x] 13. ❤️**Use appropriate properties for animations.** <br />
- [x] 14. ❤️**All element states from the style guide are implemented.** 
- [x] 15. ❤️**Links must respond to ```:hover, :active и :focus```.** <br />
- [x] 16. 💛**Interactions( ```:hover, :active и :focus```) should not change the document flow.** <br />
- [x] 17. 💛**Use vendor prefixes generated according to supported browsers.** <br />
- [ ] 18. 💛**Minify CSS files.** <br />
- [x] 19. 💛**Remove unused CSS styles.** <br />
- [x] 20. 💛**CSS should be error-free and valid.** <br />
- [ ] 21. 💚**Use CSS variables for repeated values – colors, sizes, etc.** 

1. Repeated values like colors (#0265DC, #FFFFFF) are hardcoded multiple times across files

- [ ] 22. 💚**Comment magic numbers in functions and browser hacks.** 

Images

- [x] 1. ❤️ **Correct Image Format** <br />
- [x] 2. ❤️ **Extract images without style filters from the design.** <br />
- [x] 3. ❤️ **Image Insertion** <br />
- [x] 4. ❤️ **Specify dimensions for all images.** <br />
- [x] 5. ❤️ **Provide alt text for all images.** <br />
- [x] 6. 💛 **Optimize images for browser rendering.** <br />
- [x] 7. 💚 **Use vector sprites where applicable.** <br />

Other 

- [x] 1. ❤️**The root document should have folders like css, img, js.** 
- [x] 2. ❤️ **The main page should be named index.html**
- [x] 3. ❤️ **File names and extensions should be lowercase without spaces.** <br />
- [x] 4. ❤️ **Files should be uniformly formatted.** <br />
- [ ] 5. ❤️ **The layout should match the design exactly with Perfect pixel** <br />
Layouts doesn’t match the design
<img width="1554" alt="Screenshot 2025-01-10 at 12 43 58" src="https://github.com/user-attachments/assets/b0cf2e28-76ad-4a50-baa8-5d665e35bec3" />

- [x] 6. ❤️ **The layout should display identically in the latest versions of Chrome, Opera, Firefox, Safari, and Edge.** 
- [ ] 7. ❤️ **The site should look good on all screen resolutions.** <br />
Now if width less then 1440px some elements crops
<img width="1019" alt="Screenshot 2025-01-10 at 12 47 22" src="https://github.com/user-attachments/assets/7e731176-5a28-4f67-9e8d-dc0c1baa8470" />

- [ ] 8. ❤️ **The site should handle content overflow gracefully.** <br />
Markup broke after adding more text to buttons
<img width="195" alt="Screenshot 2025-01-10 at 12 50 35" src="https://github.com/user-attachments/assets/7efd4ec3-a8aa-4bdf-8216-2b487902aed7" />
<img width="565" alt="Screenshot 2025-01-10 at 12 50 20" src="https://github.com/user-attachments/assets/61e6eeae-fcf1-4e29-9d14-eb9e2258a02f" />
<img width="151" alt="Screenshot 2025-01-10 at 12 50 01" src="https://github.com/user-attachments/assets/5ba36612-e8a0-42ae-b5bb-160970a7b974" />

- [x] 9. 💛**Unnecessary code should be removed before production.** <br />
- [ ] 10. 💚 **Use a typographer for text formatting.**  
- [x] 11. 💛 **There should be no errors in the browser console.**

Alinchik1 committed:
fix overflow, buttons, tags, header and pixels

Alinchik1 commented:
Hi Zoia,

The project is ready for the review. Fixed points from the comments:

1. Header on catalog page is divided by two, but logically it’s one h1 - fixed, one header h1
2. Layouts doesn’t match the design - fixed using Pixel Perfect Chrome plugin
3. Now if width less then 1440px some elements crops - crop fixed
4. Markup broke after adding more text to buttons - removed widths from all buttons and tags to handle content overflow


Thank you.
